src/mach/symbols.rs

```diff
@@ -125,19 +135,23 @@ pub struct Nlist {
 impl Nlist {
     /// Gets this symbol's type in bits 0xe
     pub fn get_type(&self) -> u8 {
-        self.n_type & NLIST_TYPE_MASK
+        self.n_type & N_TYPE
     }
     /// Gets the str representation of the type of this symbol
     pub fn type_str(&self) -> &'static str {
         n_type_to_str(self.get_type())
     }
     /// Whether this symbol is global or not
     pub fn is_global(&self) -> bool {
-        self.n_type & !NLIST_TYPE_MASK == NLIST_TYPE_GLOBAL
+        self.n_type & N_EXT != 0
```

Comment:
Hah, actually I changed my mind; for masking, imho, should explicitly specify what it's equal to, not that it's not equal to 0.

Reply:
Here again, I think that what I really want to test is that any bit in the mask is set. But since it's a 1 bit mask I'm fine with changing this.

Reply:
I actually don't think it's a bit mask, it's the expected value.  It's hard to tell since the original `stab.h` is pretty underdocumented.

I also think that's why I wrote the above, because the globalness of the symbol is the first 4 bits (`!NLIST_TYPE_MASK`), and the type in the remaining.

Reply:
`N_EXT` is a mask, see https://github.com/llvm-mirror/llvm/blob/120c3be2ad37c8773111b9e650ff93a97840752f/include/llvm/BinaryFormat/MachO.h#L294.

Reply:
I do not accept LLVM as an authority in this matter (actually on pretty much anything). However, [nlist.h](https://opensource.apple.com/source/xnu/xnu-344/EXTERNAL_HEADERS/mach-o/nlist.h) agrees with you.

Incidentally, this is some of the most ridiculous bit shifting shit I've ever seen take place in 8 bits, since if any bit is set in N_STAB (e.g., what you wrote, != 0), then the entire `u8` is reinterpreted via [stab.h](https://opensource.apple.com/source/xnu/xnu-3789.41.3/EXTERNAL_HEADERS/mach-o/stab.h.auto.html), (that's where your 0x64 comes from, which made no sense to me, since 4 is in the lower 4 bits and should have technically been masked out and hence irrelevant to the field):

>  * Only symbolic debugging entries have some of the N_STAB bits set and if any
>  * of these bits are set then it is a symbolic debugging entry (a stab).  In
>  * which case then the values of the n_type field (the entire field) are given
>  * in <mach-o/stab.h>